cluster/definition.go

```diff
@@ -169,6 +169,58 @@ func (d Definition) Verify() error {
 	return nil
 }
 
+func (d Definition) VerifyHashesJSON(data []byte) error {
```

Comment:
this API doesn't feel good. Needing to pass the raw bytes again doesn't feel intuitive.

I would go for something like:
```
var d Definition
err = json.Unmarshal(rawBytes, &d)

err = d.VerifyHashes()
```

For this to work, you need to unmarshal the hashes into private unexported fields so you can calculate the actual hashes and compare.

Reply:
i was also thinking the same to include this as private fields but was hesitant to do any changes in Definition type

Reply:
for this to work we would need to do json marshal every time when new Definition is called

Reply:
suggest: keep `VerifyHashesJSON(data []byte) error` but as stateless function which would introduce two function parameters as `VerifyHashesJSON(rawBytes, configHash, defHash []byte) error` ?